src/logic/schedule-exporter/workers-absence-export.logic.ts

```diff
@@ -0,0 +1,111 @@
+  }
+
+  public setWorkersWorkSheet(workSheet: xlsx.Worksheet): void {
+    workSheet.pageSetup.showGridLines = true;
+    workSheet.pageSetup.fitToPage = true;
+    workSheet.pageSetup.fitToHeight = 1;
+    workSheet.pageSetup.fitToWidth = 1;
+    workSheet.pageSetup.horizontalCentered = true;
+
+    const workersInfoArray = WorkersAbsenceExportLogic.createWorkersInfoSection(
+      this.scheduleModel,
+      this.revision
+    );
+
+    const colLens = workersInfoArray[0].map((_, colIndex) =>
+      Math.max(...workersInfoArray.map((row) => row[colIndex]?.toString().length ?? 0))
+    );
+
+    workSheet.addRows(workersInfoArray);
+
+    colLens.forEach((len, id) => {
+      workSheet.getColumn(id + 1).width = len + CELL_MARGIN;
+    });
+
+    workSheet.getColumn(1).alignment = { vertical: "middle", horizontal: "left" };
+    for (let i = 1; i <= ABSENCE_HEADERS.length; i++) {
+      workSheet.getColumn(i).alignment = { vertical: "middle", horizontal: "center" };
+    }
+
+    workSheet.getRow(1).alignment = { vertical: "middle", horizontal: "center" };
+    workSheet.getRow(1).font = { bold: true };
+  }
+
+  private static createWorkersInfoSection(
+    scheduleModel: ScheduleDataModel,
+    revision: PrimaryMonthRevisionDataModel
+  ): (string | number)[][] {
+    const names = Object.keys(scheduleModel.employee_info?.type);
+    const month = scheduleModel.schedule_info.month_number;
+    const year = scheduleModel.schedule_info.year;
+    const workers: (string | number)[][] = [];
+
+    const time = scheduleModel.employee_info.time;
+
+    workers.push(ABSENCE_HEADERS);
+    workers.push(EMPTY_ROW);
+    names.forEach((name) => {
+      const workerShifts = scheduleModel.shifts[name] as ShiftCode[];
+
+      const workerHours = WorkerHourInfo.fromWorkerInfo(
+        workerShifts,
+        revision.shifts[name] as MonthDataArray<ShiftCode>,
+        time[name],
+        scheduleModel.employee_info.contractType[name],
+        month,
+        year,
+        scheduleModel.month_info.dates,
+        scheduleModel.shift_types
+      );
+
+      workerShifts.forEach((shift, index) => {
+        let daysNo = 1;
+        if (shift !== ShiftCode.W && !SHIFTS[shift].isWorkingShift) {
+          const from = scheduleModel.month_info.dates[index];
+          while (
+            index + 1 < workerShifts.length &&
+            workerShifts[index + 1] === workerShifts[index]
+          ) {
+            index++;
+            daysNo++;
+            workerShifts[index - 1] = ShiftCode.W;
+          }
+          workerShifts[index] = ShiftCode.W;
+          workers.push([
+            name,
+            SHIFTS[shift].name,
+            `${from} ${TranslationHelper.polishMonthsGenetivus[month]}`,
+            `${scheduleModel.month_info.dates[index]} ${TranslationHelper.polishMonthsGenetivus[month]}`,
+            daysNo,
+            Math.round(
+              (daysNo * workerHours.workerHourNorm) / scheduleModel.month_info.dates.length
+            ),
```

Comment:
Tak zrozumiałam to, co jest [tutaj](https://firebasestorage.googleapis.com/v0/b/tara-cloud.appspot.com/o/orgs%2Fnurse-scheduling%2Ftasks%2F449%2Fimage-7511d6f1-7b7c-4134-a698-4030792b0cee.png?alt=media&token=8fc7466e-2470-4229-80c6-b6913791a276) i pierwszy z brzegu internetowy artykuł, ale nie jestem pewna, czy dobrze liczę godziny urlopu. Jak ktoś wie, niech mnie poprawi.